config/prow/cluster/ingress-nginx/ingress-nginx-deployment.yaml

```diff
@@ -0,0 +1,856 @@
+    type: RollingUpdate
+  minReadySeconds: 0
+  template:
+    metadata:
+      labels:
+        app.kubernetes.io/name: ingress-nginx
+        app.kubernetes.io/instance: ingress-nginx
+        app.kubernetes.io/component: controller
+    spec:
+      dnsPolicy: ClusterFirst
+      containers:
+        - name: controller
+          image: "k8s.gcr.io/ingress-nginx/controller:v1.1.1@sha256:0bc88eb15f9e7f84e8e56c14fa5735aaa488b840983f87bd79b1054190e660de"
+          imagePullPolicy: IfNotPresent
+          lifecycle: 
+            preStop:
+              exec:
+                command:
+                - /wait-shutdown
+          args:
+            - /nginx-ingress-controller
+            - --default-backend-service=$(POD_NAMESPACE)/ingress-nginx-defaultbackend
+            - --publish-service=$(POD_NAMESPACE)/ingress-nginx-controller
+            - --election-id=ingress-controller-leader
+            - --controller-class=k8s.io/ingress-nginx
+            - --configmap=$(POD_NAMESPACE)/ingress-nginx-controller
+            - --validating-webhook=:8443
+            - --validating-webhook-certificate=/usr/local/certificates/cert
+            - --validating-webhook-key=/usr/local/certificates/key
+            - --watch-ingress-without-class=true
+          securityContext: 
+            capabilities:
+              drop:
+              - ALL
+              add:
+              - NET_BIND_SERVICE
+            runAsUser: 101
+            allowPrivilegeEscalation: true
+          env:
+            - name: POD_NAME
+              valueFrom:
+                fieldRef:
+                  fieldPath: metadata.name
+            - name: POD_NAMESPACE
+              valueFrom:
+                fieldRef:
+                  fieldPath: metadata.namespace
+            - name: LD_PRELOAD
+              value: /usr/local/lib/libmimalloc.so
+          livenessProbe: 
+            failureThreshold: 5
+            httpGet:
+              path: /healthz
+              port: 10254
+              scheme: HTTP
+            initialDelaySeconds: 10
+            periodSeconds: 10
+            successThreshold: 1
+            timeoutSeconds: 1
+          readinessProbe: 
+            failureThreshold: 3
+            httpGet:
+              path: /healthz
+              port: 10254
+              scheme: HTTP
+            initialDelaySeconds: 10
+            periodSeconds: 10
+            successThreshold: 1
+            timeoutSeconds: 1
+          ports:
+            - name: http
+              containerPort: 80
+              protocol: TCP
+            - name: https
+              containerPort: 443
+              protocol: TCP
+            - name: webhook
+              containerPort: 8443
+              protocol: TCP
+          volumeMounts:
+            - name: webhook-cert
+              mountPath: /usr/local/certificates/
+              readOnly: true
+          resources: 
+            requests:
+              cpu: 100m
+              memory: 90Mi
```

Comment:
Can we give this one a bit more memory?
```
$ k -n ingress-nginx top po
NAME                                           CPU(cores)   MEMORY(bytes)
ingress-nginx-controller-75d9bf7d6b-rlp9c      3m           109Mi
ingress-nginx-controller-75d9bf7d6b-z58wp      3m           110Mi
ingress-nginx-defaultbackend-8d9b797b7-2xbz2   1m           4Mi
```
And maybe add limits as well.

Reply:
I increased the memory request of ingress-controller.
However, the authors of the nginx helm chart recommend [not setting a resource limit](https://github.com/kubernetes/ingress-nginx/blob/d16e0dec4865cbd5b937dbffb957423c308331c5/charts/ingress-nginx/values.yaml#L343-L346) and I don't know better than they do

Reply:
Hmm, what I read from the referenced blog post:

> If you are using Kubernetes, use whole CPU quotas, as this decreases the number of schedulable CPUs available to the cgroup

So what do you think about the following limits: `cpu: 1, memory: 500Mi`?

Setting limits generally makes sense to avoid pods that are too resource hungry or leaking memory to bring down other workload or even system components on the same node. 
This is a common issue that we have observed quite often in the past. Hence, I would still set limits for these components. :)

Reply:
ok, let's go for it 😄